Framework/API/src/ExperimentInfo.cpp

```diff
@@ -177,6 +185,24 @@ void checkDetectorInfoSize(const Instrument &instr,
                              "instrument");
 }
 
+boost::shared_ptr<Beamline::ComponentInfo>
+makeComponentInfo(const Instrument &oldInstr,
+                  const API::DetectorInfo &detectorInfo,
+                  std::vector<Geometry::ComponentID> &componentIds) {
+
+  InfoComponentVisitor visitor(
+      detectorInfo.size(),
+      std::bind(&DetectorInfo::indexOf, &detectorInfo, std::placeholders::_1));
+
+  // Register everything via visitor
+  oldInstr.registerContents(visitor);
```

Comment:
Have you tried using the bare instrument (without parameter map) here? That might be faster, since `IComponent::getChildren` will then not need to do allocation via `ParComponentFactory`.